From 1e9962be27718171e9291c1b62983229ca49b907 Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Wed, 18 Jan 2017 14:22:43 -0600
Subject: [PATCH] replace ${basearch} when updating the ref

pungi-make-ostree has to run on the target arch so that rpm
scriptlets can be ran.as a reult we can ask rpm what the
basearch is for the running environment

Signed-off-by: Dennis Gilmore <dennis@ausil.us>
---
 pungi/ostree/tree.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pungi/ostree/tree.py b/pungi/ostree/tree.py
index 68b6d70..94a0be2 100644
--- a/pungi/ostree/tree.py
+++ b/pungi/ostree/tree.py
@@ -16,6 +16,7 @@
 
 import os
 import json
+import rpmUtils.arch
 from kobo import shortcuts
 
 from pungi.util import makedirs
@@ -67,6 +68,8 @@ class Tree(OSTree):
             return
         ref = get_ref_from_treefile(self.treefile)
         commitid = get_commitid_from_commitid_file(self.commitid_file)
+        if ref is not None:
+            ref = ref.replace('${basearch}', rpmUtils.arch.getBaseArch())
         print('Ref: %r, Commit ID: %r' % (ref, commitid))
         if ref and commitid:
             print('Updating ref')
-- 
2.11.0

